packages/core/src/common/keybinding.ts

```diff
@@ -108,65 +195,77 @@ export class KeybindingRegistry {
 
     onStart(): void {
         for (const contribution of this.contributions.getContributions()) {
-            contribution.registerKeyBindings(this);
+            contribution.registerDefaultKeyBindings(this);
         }
     }
 
-    registerKeybindings(...bindings: Keybinding[]): void {
+    registerDefaultKeybindings(...bindings: Keybinding[]): void {
         for (const binding of bindings) {
-            this.registerKeyBinding(binding);
+            this.registerDefaultKeyBinding(binding);
         }
     }
 
     /**
-     * Adds a keybinding to the registry.
+     * Register a default keybinding to the registry.
      *
      * @param binding
      */
-    registerKeyBinding(binding: Keybinding) {
-        const existing = this.keybindings[binding.keyCode.keystroke];
-        if (existing) {
-            const collided = existing.filter(b => b.context === binding.context);
+    registerDefaultKeyBinding(binding: Keybinding) {
+        const existingBindings = this.keymapsIndex.getKeyBindingsForKeyCode(binding.keyCode);
+        if (existingBindings.length > 0) {
+            const collided = existingBindings.filter(b => b.context === binding.context);
             if (collided.length > 0) {
                 this.logger.warn(`Collided keybinding is ignored; `, JSON.stringify(binding), ' collided with ', collided.map(b => JSON.stringify(b)).join(', '));
                 return;
             }
         }
-        const { keyCode, commandId } = binding;
-        const bindings = this.keybindings[keyCode.keystroke] || [];
-        bindings.push(binding);
-        this.keybindings[keyCode.keystroke] = bindings;
-
-        const commands = this.commands[commandId] || [];
-        commands.push(binding);
-        this.commands[commandId] = bindings;
+        this.keymapsIndex.setParentKeyBinding(binding);
     }
 
     /**
+     * Override a keybinding.
+     *
+     * @param binding
+     */
+    overrideKeyBinding(binding: Keybinding) {
```

Comment:
Looks like this method would check both indexes, but should not since, I want to 'override' the parent index.

Reply:
Couldn't this override logic go into `keymapsIndex.setKeyBinding`?